pipes-and-filters/Pipes-And-Filters/Enricher/src/main/java/Enricher.java

```diff
@@ -5,54 +5,34 @@
 import java.util.function.Function;
 
 public class Enricher {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
         IAmAnOperation<Greeting, EnrichedGreeting> greetingEnricher = new GreetingEnricher();
 
         Function<String, Greeting> greetingDeserializer = messageBody -> {
             try {
                 return new ObjectMapper().readValue(messageBody, Greeting.class);
             } catch (IOException e) {
-                e.printStackTrace();
-                return null;
+                throw new RuntimeException(e);
```

Comment:
Why would we rethrow here?

Reply:
Yeah that's fair. But if you really want to take that route, you'd want the consumption, logging & counting of failed of these failures to happen in the code that _calls_ the serializer/deserializer, rather than in the ser/des code itself. You'd have to make a new interface similar to java.util.function.Function, but allow it to throw an exception (maybe make an explicit, special exception type). But I agree that may be overkill and for this exercise we want to keep it simple and just log here.

Reply:
In fact, if you look in the DataTypeChannelConsumer, the message won't end up on the invalid message queue if the deserializer just returns null. Anyway I've created another separate PR to better explain my suggestion.

Reply:
The thing I want to caution here is that this is not production messaging code. I make that point repeatedly on the course. If you want that, you use Spring or Brighter. The intent here is to build up the simple messaging tool, to let you understand the types of concern you need to have, not to provide an example you reuse in your day-to-day work